### DISCUSSION TITLE
Balkonkraftwerk Leistung integrieren/ Anschlussvarianten

### DISCUSSION THREAD

**Discussion by reimuell** [solectrus/solectrus#2862] [Q&A]
Hallo, ich plane meine Solaranlage (Senec V3) mit einem Balkonkraftwerk zu erweitern. Anschluss über eine "normale" oder intelligente Steckdose die den Ertrag messen könnte. Kann mir jemand eine Steckdose empfehlen die ich dann (wie) in Solectrus als Erzeuger integrieren kann. Brauche ich dann noch Zusatzsoftware mqtt iobroker? ich hätte es gerne "keep it simple". Oder lässt sich ein (bestimmter) Wechselrichter leichter einbinden. Ich bin noch frei in der Wahl, habe noch kein Kraftwerk gekauft.

**Comment by ledermann** [comment]
Die Unterstützung für Balkonkraftwerke wird gerade hier diskutiert:
https://github.com/orgs/solectrus/discussions/2621

Wenn du ohne MQTT-Broker auskommen willst, dann benötigst du eine andere Möglichkeit, den Ertrag abzufragen. Ich habe kürzlich den [Shelly-Collector](https://github.com/solectrus/shelly-collector) erstellt, um den Stromverbrauch einer Wärmepumpe abzufragen. Der funktioniert möglicherweise auch mit einer intelligenten Steckdose von Shelly. 

Ansonsten wird vermutlich @Zeppelin-PV etwas zum Thema sagen können ;-)

**Comment by Zeppelin-PV** [comment]
Ich habe an meinem BKW einen APsystems EZ1-M laufen.
die Daten erfasse ich mit einem Shelly 3Mpro an einer Phase
der Wechselrichter kann aber local eingebunden werden, 
dieses Prozedere mache ich aber derzeit nicht weil der WR zurückgesetzt werden muss um in den local Mode zu kommen
hierzu gibt es von Sonnenladen viel interessantes:
https://github.com/SonnenladenGmbH/APsystems-EZ1-API

**Comment by Zeppelin-PV** [comment]
Ein BKW zusammen mit der großen Dachanlage in einer Instanz von Solectrus zu integrieren ist denke ich in weite Ferne gerückt.
Nun habe ich mir gedacht für das BKW eine separate Instanz laufen zu lassen (Port:3001) um mal zunächst einfach die Produktion und den Verbrauch zu visualisieren. Gedacht, getan, Konfigurator eingesetz, alles ausgefüllt - aber
Solectrus läuft, Dashboard ist da, aber keine Verbindung.
---**Nachtrag: im Konfigurator vermisse ich die Möglichkeit über den Shelly-Collector das BKW einzubinden, hier kommt nur MQTT**
So einfach ist es also doch nicht, - zumindest nicht für mich !
Das BKW speißt auf der Terrasse auf L3 ein. Auf L2 sind alle Verbraucher, wie Licht, Steckdosen, Kameras etc.  L1 ist nur für eine Grundwasserpumpe reseriviert. Diese muss ich im Auge behalten, per HA bekomme ich eine Mail wenn diese läuft.
Alle drei Phasen auf der Terrasse werden mit einem Shelly pro 3em überwacht. (192.168.1.241)
Das Kabel von der Terrasse geht mit Zwischenstopp über die Garage ins Haus.
Auch hier, in der Garage, ist ein weiterer Shelly pro 3em verbaut, der aber nur hier die zweite Phase L2 überwacht. (192.168.1.242) 
Licht, Steckdosen, Garagentorantrieb, Kompressor etc.
L1 und L3 spielen in der Garage keine Rolle. (Vielleicht tausche in diesen noch in einen Shelly pro 1PM)
Nun würde ich gerne das Ganze so haben -
Die Produktion des BKW soll als Erzeuger dargestellt werden. Die Summe von L2 Terrasse und L2 Garage als Verbraucher.
So könnte ich beurteilen, ob in der Summe das BKW in der Lage ist Terrasse und Garage zu versorgen.
Wenn die zwei L2 Verbräuche nicht groß sind und das BKW viel produziert wird das eingespeißt, zumindest bis zum Haus.

Kann mir jemand aus der Solectrus-Gemeinde hier weiterhelfen wie so etwas im Konfigurator zu handhaben ist.
Den Verbrauch aus zweierlei Quellen zusammen zu führen denk ich, ist ein Problem.
Die zwei Verbräuche einzeln darzustellen wäre eventuell auch eine Alternative. - mit den entsprrechenden Texten.
Wenn auch die Grundwasserpumpe als Verbraucher darzustellen wäre, wäre das natürlich die Krone obendrauf.

Ich habe für interessierte Mitdenker eine Skizze damit man sich vorstellen kann wie so etwas zu realisieren wäre.
![grafik](https://github.com/user-attachments/assets/18640770-a2de-4508-bf3a-7b8718198ecf)

**Reply by ledermann** [reply]
Auch wenn ich das Szenario nicht vollständig verstanden habe, hier ein paar Hinweise, mit denen du vielleicht weiter kommst:

- Der Konfigurator deckt nicht alle erdenklichen Szenarien ab. Beispielsweise wird davon ausgegangen, dass ein Shelly ausschließlich zur Verbrauchsmessung eingesetzt wird. Daher will der Konfigurator MQTT-Einstellungen haben, wenn außer einem Wechselrichter kein Gerät gewählt wurde. In deinem Fall scheinst du auch die Erzeugung mit einem Shelly zu überwachen, diese Möglichkeit kennt der Konfigurator nicht.
- In einem so speziellen Szenario liefert der Konfigurator also bestenfalls eine Grundlage, auf der man aufbauen kann. Du musst die generierten Konfigurationsdateien vollständig verstehen, anpassen und erweitern können. Das ist keine Standardsituation, die mit Copy/Paste lösbar wäre.
- Grundsätzlich: Wenn du zwei Shelly hast, brauchst du zwei Shelly-Collectoren, für jede IP einen eigenen. Der Shelly-Collector fragt die Phasen einzeln (`power_a`, `power_b`, `power_c`) und gesamt (`power`) ab. Du kannst die Sensoren vom Dashboard einzelnen Phasen zuweisen (z.B. `INFLUX_SENSOR_INVERTER_POWER=garage:power_b`). Eine Summenbildung ausgewählter Phasen (z.B. nur a+b) gibt es aber nicht.

Es sollte also möglich sein, zumindest eine grundlegende Anzeige deines BKW in SOLECTRUS hinzubekommen. Vielleicht nur mit drei Sensoren: Erzeugung, Verbrauch, Einspeisung (ins Haus).

Bezüglich "Keine Verbindung" - ich nehme an, du meinst die Status-Anzeige oben rechts. Das ist eine sehr allgemeine Fehlerbeschreibung, aus der man überhaupt nicht auf die Ursache schließen kann. Das bedeutet nur, dass keine aktuellen Messwerte gefunden werden. Schau dir die Logs an (sehr wichtig!) und prüfe, ob die Sensoren und Collectoren so konfiguriert sind, dass es überhaupt funktionieren kann. Das Grundprinzip ist immer: Ein Collector sammelt Messwerte ein und schreibt sie nach InfluxDB in ein gewähltes Measurement/Field. Das Dashboard holt sie dann dort raus und stellt sie dar. Ein häufiger Fehler ist, dass der Collector Messwerte nach X schreibt, das Dashboard aber in Y sucht.

**Reply by Zeppelin-PV** [reply]
DANKE Georg
momentan ist es mir nicht vergönnt die nötige Zeit in meine Möchtegern-Vorhaben zu stecken.
Ich melde mich irgendwann wieder wenn ich mit dem BKW doch weitermachen will.
LG Ernst

**Comment by ledermann** [comment]
Ich habe jetzt mal ein Issue angelegt: solectrus/solectrus#3854
